pkg/frontend/admin_openshiftcluster_kubernetespods_logs_test.go

```diff
@@ -0,0 +1,194 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	mock_adminactions "github.com/Azure/ARO-RP/pkg/util/mocks/adminactions"
+)
+
+func TestAdminKubernetesGetPodLogs(t *testing.T) {
```

Comment:
I have added few more validation tests in `TestValidateAdminKubernetesPodLogs`.

Regarding `KubeGetPodLogs` returning an error, I don't think it will return any error in unit test as there is the kubernetes client library communicating to the kubernetes server to get logs .. this won't see an error in unit tests .. but we do have e2e test for this.

Reply:
> Regarding `KubeGetPodLogs` returning an error, I don't think it will return any error in unit test as there is the kubernetes client library communicating to the kubernetes server to get logs .. this won't see an error in unit tests .. but we do have e2e test for this.

In real world `KubeGetPodLogs` can return an error. For example, if something is wrong with an API server. We need to emulate this situation in unit tests.

I haven't looked at e2es just yet.